server/Arcadia.Assistant.Health/HealthActor.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace Arcadia.Assistant.Health
+{
+    using System;
+    using System.Data;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    using Abstractions;
+    using Akka.Actor;
+    using CSP;
+    using CSP.Model;
+    using CSP.Vacations;
+    using Microsoft.EntityFrameworkCore;
+
+    public class HealthActor : UntypedActor, ILogReceive
+    {
+        private readonly Func<ArcadiaCspContext> contextFactory;
+        private readonly Remote1CConfiguration bkConfiguration;
+
+        public HealthActor(Func<ArcadiaCspContext> contextFactory, Remote1CConfiguration bkConfiguration)
+        {
+            this.contextFactory = contextFactory;
+            this.bkConfiguration = bkConfiguration;
+        }
+
+        protected override void OnReceive(object message)
+        {
+            if (!(message is HealthCheckMessage healthMessage))
+            {
+                this.Unhandled(message);
+                return;
+            }
+
+            switch (healthMessage.CheckType)
+            {
+                case HealthCheckType.Server:
+                    this.Sender.Tell(true);
+                    break;
+
+                case HealthCheckType.Check1C:
+                    this.Check1C().PipeTo(this.Sender);
+                    break;
+
+                case HealthCheckType.CheckDatabase:
+                    this.Sender.Tell(this.CheckDatabase());
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private async Task<bool> Check1C()
+        {
+            try
+            {
+                var uriBuilder = new UriBuilder(this.bkConfiguration.Url);
+
+                var client = new EnterpriseV8(uriBuilder.Uri)
```

Comment:
A proper way to do that would be to store Status somewhere in the actor that already talks to database / 1C. When just ask it to return its status.

Reply:
Also minus is that currently period of refresh is once in 10 minutes. It is not frequent enough.

Reply:
Kept as is - all checks are executed inside health actor. Reasons were listed above.